braindecode/models/conformer.py

```diff
@@ -0,0 +1,279 @@
+
+    This neural network architecture recieves a different input shape than
+    other models. The input shape should be four-dimensional matrix
+    representing the EEG signals.
+
+         (batch_size, 1, n_channels, n_timesteps)`.
+
+    The EEG Conformer architecture is composed of three modules:
+        - PatchEmbedding
+        - TransformerEncoder
+        - ClassificationHead
+
+    Notes
+    -----
+    The authors recommend using augment data before using Conformer, e.g. S&R,
+    at the end of the code.
+    Please refer to the original paper and code for more details.
+
+    .. versionadded:: 0.7.1
+
+    We aggregate the parameters based on the parts of the models.
+
+    Parameters PatchEmbedding
+    -------------------------
+    - n_kernel: int
+        Number of kernels for the temporal convolution layer (first layer).
+    - n_kernel_temp_conv: int
+        Kernel size of the temporal convolution layer.
+    - eeg_channel: int
+        Number of EEG channels (kernel size of the spatial convolution layer).
+    - n_kernel_avg_pool: int
+        Kernel size of the average pooling layer.
+    - stride_avg_pool: int
+        Stride of the average pooling layer.
+    - drop_prob: float
+        Dropout rate of the convolutional layer.
+
+    Parameters TransformerEncoder
+    -----------------------------
+    - att_depth: int
+        Number of self-attention layers.
+    - att_heads: int
+        Number of attention heads.
+    - att_drop: float
+        Dropout rate of the self-attention layer.
+
+    Parameters ClassificationHead
+    -----------------------------
+    - fc_dim: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes.
+
+
+    References
+    ----------
+    .. [EEG Conformer] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [EEG Conformer Code] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+
+
+    """
+
+    def __init__(
+            self,
+            kernel=40,
+            kernel_temp_conv=25,
+            eeg_channel=22,
+            kernel_avg_pool=75,
+            stride_avg_pool=15,
+            conv_drop=0.5,
+            att_depth=6,
+            att_heads=10,
+            att_drop=0.5,
+            fc_dim=2440,
+            n_classes=4,
+    ):
+        super().__init__(
+            PatchEmbedding(
+                kernel,
+                kernel_temp_conv,
+                eeg_channel,
+                kernel_avg_pool,
+                stride_avg_pool,
+                conv_drop,
+            ),
+            TransformerEncoder(att_depth, kernel, att_heads, att_drop),
```

Comment:
As mentioned below, use the official torch implementations instead:
```suggestion
            nn.TransformerEncoder(nn.TransformerEncoderLayer(
            d_model=kernel, nhead=att_heads, dim_feedforward=kernel*4, dropout=att_drop, 
            batch_first=True, norm_first=True, activation="gelu"), num_layers=att_depth)
```

Reply:
I implemented the Transformer part in my previous study to adjust it easier, instead of using the official torch version. Please wait for me a little time to check.